doc/ci_admin_guide/Docker_registry_provisioning.md

```diff
@@ -0,0 +1,152 @@
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-nova-consoleauth:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-nova-ssh:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-nova-scheduler:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-keystone:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-neutron-server:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-neutron-metadata-agent:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-swift-account:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-swift-object:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-swift-rsyncd:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-swift-object-expirer:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-swift-proxy-server:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-swift-container:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-barbican-api:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-barbican-keystone-listener:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-barbican-worker:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-heat-api:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-heat-api-cfn:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-heat-engine:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-horizon:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-swift-base:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-memcached:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-cron:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-rabbitmq:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-kolla-toolbox:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-nova-libvirt:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-fluentd:ocata
+ci-repo.englab.juniper.net:5000/kolla/centos-binary-mariadb:ocata
+ci-repo.englab.juniper.net:5000/contrail-openstack-neutron-init:ocata-master-91
+ci-repo.englab.juniper.net:5000/contrail-openstack-compute-init:ocata-master-91
+```
+</details>
+
+Then, copy this file to the registry, and pull the images:
+
+```bash
+for image in `cat pullme`; do docker pull $image; done
+```
+
+### Starting the registry
+
+> NOTE: This section is adapted from [Docker registry deploying instruction][docker-registry-deploying].
+
+First, start the registry container:
+
+```bash
+docker run -d -p 5000:5000 --restart=always --name registry registry:2
+```
+
+Then, we need to tag the images we've pulled and push them to the registry:
+
+Kolla images:
+
+```bash
+for image in $(docker images | grep ci-rep | grep kolla | awk '{print $1}'); do
+    name=$(echo $image | cut -f2- -d/);
+    docker tag $image:ocata localhost:5000/$name:ocata;
+    docker push localhost:5000/$name;
+done
+```
+
+Other images:
+
+```bash
+for image in $(docker images | grep ci-rep | grep ocata-master | awk '{print $1}'); do
```

Comment:
How do we know when `ocata-master` should be changed to some other value? And how to determine new value? Why `-91`?

Reply:
Yes, the scripts are hardcoded for the specific version (as it's mentioned in TODO). Perhaps we should add a note that `ocata-master` etc. should be substituted in these scripts? I would like to make the scripts more robust, but I think it could stay as is (with a note) and make them fixed properly when writing proper ansible scripts. What do you think?

Reply:
We can leave `ocata-master` here, but it would be great to have a few words about meaning of this fixed string.